File: 0003 - Longest Substring/0003.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        longest = ''
        for i, char in enumerate(s):
            found = []
            sub = ''
            for c in s[i:]:
                if c in found:
                    break
                found.append(c)
                sub += c
            if len(sub) > len(longest):
                longest = sub
        return len(longest)

File: 0003 - Longest Substring/0003.go
func lengthOfLongestSubstring(s string) int {
	longest := ""
	for i, _ := range s {
		found := make(map[string]int)
		sub := ""
		for _, c := range s[i:] {
			d := string(c)
			if found[d] == 1 {
				break
			}
			found[d] += 1
			sub += d
		}
		if len(sub) > len(longest) {
			longest = sub
		}
	}
	return len(longest)
}